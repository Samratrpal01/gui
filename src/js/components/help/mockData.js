export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master'
};
