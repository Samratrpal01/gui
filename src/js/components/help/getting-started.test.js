import React from 'react';

import { undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import GettingStarted from './getting-started';
import { helpProps } from './mockData';

describe('GettingStarted Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<GettingStarted {...helpProps} />);
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
