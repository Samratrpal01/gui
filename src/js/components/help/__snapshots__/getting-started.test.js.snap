// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GettingStarted Component renders correctly 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.25rem;
}

<div>
  <h2>
    Getting started
  </h2>
  <h3>
    Get Started guide
  </h3>
  <p>
    To quickly get started with the common use cases of Mender, follow the
     
    <a
      href="https://docs.mender.io/get-started"
      rel="noopener noreferrer"
      target="_blank"
    >
      Get Started chapter in our documentation 
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-0"
        data-testid="LaunchIcon"
        focusable="false"
        style="vertical-align: text-bottom;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
    .
  </p>
  <h3>
    Installing Mender on your devices
  </h3>
  <p>
    We support several ways of integrating your devices with Mender. For a detailed introduction read the
     
    <a
      href="https://docs.mender.io/client-installation"
      rel="noopener noreferrer"
      target="_blank"
    >
      Client Installation documentation 
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-0"
        data-testid="LaunchIcon"
        focusable="false"
        style="vertical-align: text-bottom;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
    .
  </p>
  <p>
    Integrations for different boards and operating systems can be found on
     
    <a
      href="https://hub.mender.io/c/board-integrations/6"
      rel="noopener noreferrer"
      target="_blank"
    >
      Mender Hub 
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-0"
        data-testid="LaunchIcon"
        focusable="false"
        style="vertical-align: text-bottom;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
  </p>
  <h3>
    Creating Artifacts
  </h3>
  <p>
    Everything you need to package your software for deployment with Mender can be found here:
     
    <a
      href="https://docs.mender.io/artifact-creation"
      rel="noopener noreferrer"
      target="_blank"
    >
      Artifact Creation documentation 
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall emotion-0"
        data-testid="LaunchIcon"
        focusable="false"
        style="vertical-align: text-bottom;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
    .
  </p>
</div>
`;
